sdk/tables/Azure.Data.Tables/src/Sas/TableAccountSasBuilder.cs

```diff
@@ -0,0 +1,233 @@
+            var stringToSign = string.Join("\n",
+                sharedKeyCredential.AccountName,
+                Permissions,
+                TableConstants.Sas.TableAccountServices.Table,
+                ResourceTypes.ToPermissionsString(),
+                startTime,
+                expiryTime,
+                IPRange.ToString(),
+                Protocol.ToProtocolString(),
+                Version,
+                "");
+            var signature = TableSharedKeyCredential.ComputeSasSignature(sharedKeyCredential, stringToSign);
+            var p = new TableAccountSasQueryParameters(
+                version: Version,
+                services: TableAccountSasServices.Tables,
+                resourceTypes: ResourceTypes,
+                protocol: Protocol,
+                startsOn: StartsOn,
+                expiresOn: ExpiresOn,
+                ipRange: IPRange,
+                identifier: Identifier,
+                resource: null,
+                permissions: Permissions,
+                signature: signature);
+            return p;
+        }
+
+        /// <summary>
+        /// Use an account's <see cref="TableSharedKeyCredential"/> to sign this
+        /// shared access signature values to produce the proper SAS query
+        /// parameters for authenticating requests.
+        /// </summary>
+        /// <param name="sharedKeyCredential">
+        /// The storage account's <see cref="TableSharedKeyCredential"/>.
+        /// </param>
+        /// <returns>
+        /// A URL encoded query string representing the SAS.
+        /// </returns>
+        public string Sign(TableSharedKeyCredential sharedKeyCredential) =>
+            ToSasQueryParameters(sharedKeyCredential).ToString();
+
+        /// <summary>
+        /// Returns a string that represents the current object.
+        /// </summary>
+        /// <returns>A string that represents the current object.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override string ToString() => base.ToString();
+
+        /// <summary>
+        /// Check if two TablesSasBuilder instances are equal.
+        /// </summary>
+        /// <param name="obj">The instance to compare to.</param>
+        /// <returns>True if they're equal, false otherwise.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override bool Equals(object obj) => base.Equals(obj);
+
+        /// <summary>
+        /// Get a hash code for the TablesSasBuilder.
+        /// </summary>
+        /// <returns>Hash code for the TablesSasBuilder.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override int GetHashCode() => base.GetHashCode();
+
+        /// <summary>
+        /// Ensure the <see cref="TableAccountSasBuilder"/>'s properties are in a
+        /// consistent state.
+        /// </summary>
+        private void EnsureState()
```

Comment:
This should probably validate that all required properties are set. I don't think the checks we have here now are correct for account level SAS. [See this](https://docs.microsoft.com/en-us/rest/api/storageservices/create-account-sas#specifying-account-sas-parameters) for which properties are required.

Reply:
Made the changes, but happened to notice that Storage didn't use an `EnsureState()` for their [account-level](https://github.com/meanjeekim/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Common/src/Sas/AccountSasBuilder.cs#L115) but did for [service-level SAS](https://github.com/meanjeekim/azure-sdk-for-net/blob/31d0c5432797a5dba4048c1040cfd8b295335adc/sdk/storage/Azure.Storage.Blobs/src/Sas/BlobSasBuilder.cs#L270). Wondering why one put the property checks in a method versus not, other than maybe the complexity of the if-statements?